File: example/ios/Runner/PhotoDetail.swift
//
//  PhotoDetail.swift
//  Sorted
//
//  Created by Grigori on 3/3/20.
//  Copyright © 2020 Sorted. All rights reserved.
//

import Foundation
import RealmSwift

public final class PhotoDetail: Object {
    @objc dynamic var _id: String = ""

    /// This parameters must be calculated relatively
    /// ``` centerx = realCenterx / containerWidth ```
    @objc dynamic var centerx: Double = 0
    
    /// This parameters must be calculated relatively
    /// ``` centery = realCentery / containerWidth ```
    @objc dynamic var centery: Double = 0
    
    @objc dynamic var rotate: Double = 0
    @objc dynamic var timestamp: Double = 0
    @objc dynamic var zoom: Double = 1

    override public class func primaryKey() -> String? {
        return "_id"
    }
}

File: example/lib/photo_detail.dart
import 'package:flutterrealm_light/object.dart';

class PhotoDetail extends RLMObject {
  String id = '';

  /// This parameters must be calculated relatively
  /// ``` centerx = realCenterx / containerWidth ```
  double centerx = 0;

  /// This parameters must be calculated relatively
  /// ``` centery = realCentery / containerHeight ```
  double centery = 0;

  double rotate = 0;
  double timestamp = 0;
  double zoom = 1;

  @override
  PhotoDetail fromJson(Map json) {
    id = json['_id'];

    centerx = json['centerx'];
    centery = json['centery'];
    rotate = json['rotate'];
    timestamp = json['timestamp'];
    zoom = json['zoom'];

    return this;
  }

  @override
  Map toJson() {
    Map map = super.toJson();
    map['_id'] = id;
    map['centerx'] = centerx;
    map['centery'] = centery;
    map['rotate'] = rotate;
    map['timestamp'] = timestamp;
    map['zoom'] = zoom;
    return map;
  }
}